main.py

```diff
@@ -413,123 +508,130 @@ def performEvaluation(issue_data) -> Result:
 
     for line_no in cf_crash_line: # if multiple crash location found, one shoud match exactly with the expected crash information
         crashed_class_name_line = -1
         for i in range(line_no, line_no + 5): # should be immediate next line of crash line
-            if log_file_data_list[i].strip().startswith("Compilation unit:"):
+            if lines_of_logs[i].strip().startswith("Compilation unit:"):
                 crashed_class_name_line = i
                 break
         if crashed_class_name_line == -1:
             continue  # start looking for next crash location
-        class_name_abs_path = log_file_data_list[crashed_class_name_line].split(" ")[-1]
+        class_name_abs_path = lines_of_logs[crashed_class_name_line].split(" ")[-1]
         crashed_class_name = os.path.basename(class_name_abs_path)
 
         exception_line = -1
         for i in range(crashed_class_name_line, crashed_class_name_line + 5): # should be immediate next line of crash line
-            if log_file_data_list[i].strip().startswith("Exception:"):
+            if lines_of_logs[i].strip().startswith("Exception:"):
                 exception_line = i
                 break
         exception_stack = [] #compare it with actual stack trace
-        exception_line_str = log_file_data_list[exception_line] #Exception: java.lang.NullPointerException; java.lang.NullPointerException
+        exception_line_str = lines_of_logs[exception_line] #Exception: java.lang.NullPointerException; java.lang.NullPointerException
         exception_line_sub_str = (exception_line_str[exception_line_str.index("Exception:") + 10:]).split()[0] # java.lang.NullPointerException; java.lang.NullPointerException
         exception_cause = re.sub(r'^[^a-zA-Z]+|[^a-zA-Z]+$', '', exception_line_sub_str) # java.lang.NullPointerException
+        
+        if not require_stack and crashed_class_name and exception_cause: # if stack is not required, not adding them in exception data. 
+            exception_data = ExceptionData(crashed_class_name, exception_cause)
+            return_data.append(exception_data)
+            continue
+        
         for i in range(exception_line + 1, exception_line + 6):
-            if log_file_data_list[i].lstrip().startswith("at"):
-                exception_stack.append(log_file_data_list[i].split()[-1].strip())
+            if lines_of_logs[i].lstrip().startswith("at"):
+                exception_stack.append(lines_of_logs[i].split()[-1].strip())
         
-        if crashed_class_name != None and exception_cause != None and len(exception_stack) > 0:
+        if crashed_class_name and exception_cause and len(exception_stack) > 0:
             exception_data = ExceptionData(crashed_class_name, exception_cause, exception_stack)
             return_data.append(exception_data)
 
     return return_data
 
-def compare_crash_log(expected_log_path, actual_log_path):
+def compare_crash_log(expected_log_path, actual_log_path, require_stack = False):
```

Comment:
I would prefer that `require_stack` default to `True` rather than `False`, since in most cases we want to compare the stack traces; the only time we don't want to compare stack traces is StackOverflowErrors and other similar things (like OutOfMemoryErrors), which should be the minority of crashes.